BookingsApi/BookingsApi/Controllers/WorkHoursController.cs

```diff
@@ -171,11 +168,31 @@ public async Task<IActionResult> UpdateVhoNonAvailabilityHours(string username,
                 ModelState.AddFluentValidationErrors(hourValidationResult.Errors);
                 return BadRequest(ModelState);
             }
-
             var updateNonWorkingHoursCommand = new UpdateNonWorkingHoursCommand(request.Hours);
             await _commandHandler.Handle(updateNonWorkingHoursCommand);
             
             return NoContent();
         }
+        
+        /// <summary>
+        /// Delete non availability work hours for vho
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns>vho with list of non availability work hours</returns>
+        [HttpDelete("/NonAvailability")]
+        [OpenApiOperation("DeleteVhoNonAvailabilityHours")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        public async Task<IActionResult> DeleteVhoNonAvailabilityHours(long id)
+        {
+            if (id <= 0)
+            {
+                ModelState.AddModelError(nameof(id), $"Please provide a valid {nameof(id)}");
+                return BadRequest(ModelState);
+            }
```

Comment:
we should probably also return a 404 here if the non availability doesn't exist. Maybe do a `GetVhoNonAvailabilityById` query, and then return a bad request it returns null.

See this example for deleting a hearing: https://github.com/hmcts/vh-bookings-api/blob/master/BookingsApi/BookingsApi/Controllers/HearingsController.cs#L494

Reply:
not sure about this. The example you added is returning an object and if it is null is not found. 
In my case it is not returning an object so I can't check it

Reply:
refactored